validator_worker/src/sentry_interface.rs

```diff
@@ -162,47 +151,54 @@ impl<T: Adapter + 'static> SentryApi<T> {
         &self,
         after: DateTime<Utc>,
     ) -> Result<EventAggregateResponse, Box<ValidatorWorker>> {
-        let mut adapter = self
+        let auth_token = self
             .adapter
             .write()
-            .compat()
             .await
-            .expect("get_event_aggregates: Failed to acquire adapter");
-
-        let auth_token = adapter
             .get_auth(&self.whoami)
             .map_err(|e| Box::new(ValidatorWorker::Failed(e.to_string())))?;
 
         let url = format!(
             "{}/events-aggregates?after={}",
             self.validator_url,
-            after.timestamp()
+            after.timestamp_millis()
         );
 
-        let future = self
+        let result: EventAggregateResponse = self
             .client
             .get(&url)
-            .header(AUTHORIZATION, auth_token.to_string())
+            .bearer_auth(&auth_token)
             .send()
-            .and_then(|mut res: Response| res.json::<EventAggregateResponse>())
-            .map_err(|e| Box::new(ValidatorWorker::Failed(e.to_string())));
+            .map_err(|e| Box::new(ValidatorWorker::Failed(e.to_string())))
+            .compat()
+            .await?
+            .json()
+            .map_err(|e| Box::new(ValidatorWorker::Failed(e.to_string())))
+            .compat()
+            .await?;
```

Comment:
you can just remove the `;` and the `Ok(result)`?

Reply:
I felt looks much better to the eyes this way, I didn't want to have `Ok()` with the huge content within it

Reply:
You can remove the `?` as well, since `.await` will give you already a `Result` no need of `Ok(...)`